src/core/server/saved_objects/import/utils.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { parse, stringify } from 'hjson';
+import { SavedObject, SavedObjectsClientContract } from '../types';
+
+export interface UpdateDataSourceNameInVegaSpecProps {
+  spec: string;
+  newDataSourceName: string;
+  previousDataSourceName?: string;
+}
+
+export const updateDataSourceNameInVegaSpec = (
+  props: UpdateDataSourceNameInVegaSpecProps
+): string => {
+  const { spec } = props;
+  let parsedSpec = parseJSONSpec(spec);
+  const isJSONString = !!parsedSpec;
+  if (!parsedSpec) {
+    parsedSpec = parse(spec, { keepWsc: true });
+  }
+
+  const dataField = parsedSpec.data;
+
+  if (dataField instanceof Array) {
+    parsedSpec.data = dataField.map((dataObject) => {
+      return updateDataObject(dataObject, props);
+    });
+  } else if (dataField instanceof Object) {
+    parsedSpec.data = updateDataObject(dataField, props);
+  } else {
+    throw new Error(`"data" field should be an object or an array of objects`);
+  }
+
+  return isJSONString
+    ? JSON.stringify(parsedSpec)
+    : stringify(parsedSpec, {
+        bracesSameLine: true,
+        keepWsc: true,
+      });
+};
+
+export const getDataSourceTitleFromId = async (
+  dataSourceId: string,
+  savedObjectsClient: SavedObjectsClientContract
+) => {
+  return await savedObjectsClient.get('data-source', dataSourceId).then((response) => {
+    const attributes: any = response?.attributes || {};
+    return attributes ? attributes.title : undefined;
+  });
+};
+
+export const extractVegaSpecFromSavedObject = (savedObject: SavedObject) => {
+  if (isVegaVisualization(savedObject)) {
+    // @ts-expect-error
+    const visStateObject = JSON.parse(savedObject.attributes?.visState);
+    return visStateObject.params.spec;
+  }
+
+  return false;
```

Comment:
Nit:
Since the function named as `extractVegaSpecFromSavedObject`, the called of this func will be expect an object.
What if we return undefined here instead of a boolean. What do you think?

Reply:
Changed to `undefined`